lib/parser/lexer.rl

```diff
@@ -332,7 +335,9 @@ class Parser::Lexer
     end
 
     def tok(s = @ts, e = @te)
-      @source[s...e].encode(@encoding)
+      s = @source[s...e]
```

Comment:
Please do not reuse the position variable `s` for token value. Call it, well, `value`.